Add list of css features in markdown by category and propose RFC (#91)
CSS-Next/css-next: lists/proposal-rfc.md

> *Note that this is a draft / template and just a suggestion by one person for the moment, this is by no means a final RFc*
> 

Author: ```suggestion
```

# Rfc

> The term "CSS3" has been widely used to encompass various additions and enhancements to Cascading Style Sheets (CSS) since around 2010. However, as CSS continues to evolve with new features and specifications, the blanket term "CSS3" has become insufficient and misleading. This RFC proposes the categorization of CSS properties into distinct groups, namely CSS3, CSS4, CSS5, and Future/Next, to better organize and facilitate understanding of the evolving CSS landscape. This categorization aims to improve adoption and ease of teaching, while not having a direct impact towards the CSS Working Group (CSSWG) operations and specifications.

Reviewer: ```suggestion
The term "CSS3" has been widely used to encompass various additions and enhancements to Cascading Style Sheets (CSS) since around 2010. However, as CSS continues to evolve with new features and specifications, the blanket term "CSS3" has become insufficient and misleading. This RFC proposes the categorization of CSS properties into distinct groups, namely CSS3, CSS4, and CSS5, to better organize and facilitate understanding of the evolving CSS landscape. This categorization aims to improve adoption and ease of teaching, while not having a direct impact towards the CSS Working Group (CSSWG) operations or the [Baseline initiative](https://web.dev/baseline). 
```

> The term "CSS3" has been widely used to encompass various additions and enhancements to Cascading Style Sheets (CSS) since around 2010. However, as CSS continues to evolve with new features and specifications, the blanket term "CSS3" has become insufficient and misleading. This RFC proposes the categorization of CSS properties into distinct groups, namely CSS3, CSS4, and CSS5, to better organize and facilitate understanding of the evolving CSS landscape. This categorization aims to improve adoption and ease of teaching, while not having a direct impact towards the CSS Working Group (CSSWG) operations or the [Baseline initiative](https://web.dev/baseline). 

Reviewer 2: ```suggestion
The term "CSS3" has been widely used to encompass various additions and enhancements to Cascading Style Sheets (CSS) since around 2010. However, as CSS continues to evolve with new features and specifications, the blanket term "CSS3" has become insufficient and misleading. This RFC proposes the categorization of CSS properties into distinct groups, namely CSS3, CSS4, and CSS5, to better organize and facilitate understanding of the evolving CSS landscape. This categorization aims to improve adoption and ease of teaching, while not having a direct impact towards the CSS Working Group (CSSWG) operations or the [Baseline initiative](https://github.com/web-platform-dx/web-features/blob/main/docs/baseline.md). 
```

## Reason

The term "CSS3" has been used to refer to all additions and enhancements to CSS since the early 2010s. However, CSS has continued to evolve beyond the scope of what was traditionally considered CSS3, leading to confusion in recruitment profiles, teaching materials, and documentation. This lack of clear categorization makes it challenging for developers to stay updated with the latest features and for educators to effectively teach CSS advancements and define specializations in the field.

## Proposal

### Categorization

> Categorization is based on when new properties were introduced in a spec. This is to be browser(vendor) independent.

Reviewer: ```suggestion
Categorization is based on when new properties were introduced in a spec. This is browser independent and implementation independent, though these factors are lightly taken into account.
```

> Categorization is based on when new properties were introduced in a specification. This is browser independent and implementation independent, though these factors are lightly taken into account.

Reviewer 2: ```suggestion
Categorization is based on a variety of factors with the primary factor being the general timeline that an API was initially specified in the CSS Working Group. Additionally, implementer interest and adoption were moderately taken into account in the categorization process.
```

> * **CSS3:** This category includes CSS properties and features introduced prior to the proposal to further categorize. It serves as a historical reference point for CSS advancements up to a certain point.

Reviewer: ```suggestion
* **CSS3:** This category was established before this community group began articulating levels and was defined by the CSSWG, and was designed to be open ended. It serves as a historical reference point for CSS advancements up to a certain point.
```

> * **CSS4:** Properties and features introduced in a spec between 2013 - 2018.

Reviewer 2: ```suggestion
* **CSS4:** Properties and features introduced in a spec around 2013 - 2018.
```

> * **CSS5:** Properties and features introduced in a spec between 2019 - 2024.

Reviewer 2: ```suggestion
* **CSS5:** Properties and features introduced in a spec around 2019 - 2024.
```

> * **Future/Next**: This category encompasses properties and features that are currently in development or proposed for future versions of CSS beyond CSS5. It serves as a preview of upcoming CSS that are currently known.

Reviewer 2: ```suggestion
* **Future/Next**: This category is a placeholder for properties and features that are currently in development or proposed for future versions of CSS beyond CSS5.
```

### Implementation:

* Existing CSS properties and features will be retroactively categorized into CSS3, CSS4, or CSS5 based on their introduction dates in specs.

> * New CSS properties and features will be categorized into Future/Next (name pending)

Reviewer 2: ```suggestion
```

* The categorization process will be transparent and documented
* Each property has been deliberated extensively to easily fit this categorization.

> * The categorization will not affect the numbering or operations of official CSS snapshots maintained by the CSSWG.

Reviewer 2: ```suggestion
* The categorization will not affect the numbering or operations of official CSS snapshots maintained by the CSSWG and [Baseline initiative](https://github.com/web-platform-dx/web-features).
```

## Benefits

### Clarity and Organization

> Clear categorization of CSS properties into CSS3, CSS4, CSS5, and Future/Next will provide developers, recruiters, and educators with a structured framework for understanding and discussing CSS advancements and skill level.

Reviewer 2: ```suggestion
"Modern CSS" does not have any specific meaning or timeframe. Clear categorization of CSS properties into CSS3, CSS4, CSS5, and any future versions will provide developers, recruiters, employers, and educators with a structured framework for understanding and discussing CSS advancements and its evolution.
```

### Improved Adoption
By simplifying the presentation of CSS advancements, developers may be more inclined to explore and adopt new features, leading to broader utilization of modern CSS capabilities.

### Enhanced Teaching

> Educators will have a clearer roadmap for teaching CSS advancements and could even be divided in different classes, making it easier to introduce concepts to students.

Reviewer 2: ```suggestion
Educators will have a clearer vocabulary and roadmap for teaching the evolution of CSS and better categorize modern CSS features.
```

## Conclusion

> The categorization of CSS properties into CSS3, CSS4, CSS5, and Future/Next offers a solution to the ambiguity surrounding the term "CSS3" and provides a structured framework for understanding and discussing CSS advancements. By implementing this proposal, we can improve adoption, teaching, and better organize the evolving landscape of CSS.

Reviewer 2: ```suggestion
The categorization of CSS properties into CSS3, CSS4, CSS5, and any future versions offers a solution to the ambiguity surrounding the terms "CSS3" and "modern CSS", while providing a structured framework for understanding and discussing CSS advancements. By implementing this proposal, we can improve adoption, simplify education, and better organize the evolving landscape of CSS.
```

* **CSS3:** This category includes CSS properties and features introduced prior to the proposal to further categorize. It serves as a historical reference point for CSS advancements up to a certain point.
* **CSS4:** Properties and features introduced in a spec between 2013 - 2018.
* **CSS5:** Properties and features introduced in a spec between 2019 - 2024.

> * **Future/Next**: This category encompasses properties and features that are currently in development or proposed for future versions of CSS beyond CSS5. It serves as a preview of upcoming CSS that are currently known.

Reviewer 3: Do we also need to mention the year here? Like "2024+"

Reviewer 2: 2025+ will be the next version (2025 - 2031 based on our current versioning)

Author: I agree, but not sure if we should add an "end" date already. As CSS features really come alive a lot more quickly the past few years. Versioning might have a need to go a bit faster than 6 years. Just a thought. (To clarify: not talking yearly, but rather "who knows, maybe 4 years")

Reviewer 2: When we used this timeline, it was more of a guide and not an absolute (and not all the features neatly fit into this definition set as there were a lot of exceptions). The phrasing here makes it sound like absolute cutoffs and like we used the dates as the primary mechanism (which was not true, it was an indicator, along with other indicators like spec level and browser support).
